python/cuml/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -648,9 +652,17 @@ class NearestNeighbors(UniversalBase,
                                              axis=1)
                 I_cparr = cp.take_along_axis(I_cparr, correct_order, axis=1)
 
-                D_ndarr = cuml.common.input_to_cuml_array(D_cparr).array
+                D_ndarr = cuml.common.input_to_cuml_array(
+                    D_cparr,
+                    convert_to_dtype=(np.float32 if convert_dtype
+                                      else None),
+                ).array
                 D_ndarr.index = index
-                I_ndarr = cuml.common.input_to_cuml_array(I_cparr).array
+                I_ndarr = cuml.common.input_to_cuml_array(
+                    I_cparr,
+                    convert_to_dtype=(np.float32 if convert_dtype
```

Comment:
This looks incorrect. Default should be `np.int32` or `np.int64`, I'm not sure what the API accepts

Reply:
good catch, this should be int64